auth/services/selfsigned/signer.go

```diff
@@ -29,30 +29,62 @@ import (
 	"github.com/nuts-foundation/go-did/vc"
 	"github.com/nuts-foundation/nuts-node/auth/contract"
 	"github.com/nuts-foundation/nuts-node/auth/services"
+	"github.com/nuts-foundation/nuts-node/auth/services/selfsigned/types"
+	"github.com/nuts-foundation/nuts-node/auth/services/selfsigned/web"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/vcr"
 	"github.com/nuts-foundation/nuts-node/vcr/credential"
 	"github.com/nuts-foundation/nuts-node/vcr/holder"
 	"github.com/nuts-foundation/nuts-node/vcr/signature/proof"
+	"net/url"
 	"time"
 )
 
 const credentialType = "NutsEmployeeCredential"
 
-func (v service) SigningSessionStatus(ctx context.Context, sessionID string) (contract.SigningSessionResult, error) {
-	s, ok := v.sessions[sessionID]
+// signer implements the contract.Signer interface
+type signer struct {
+	store     types.SessionStore
+	vcr       vcr.VCR
+	publicURL string
+	// The time the user has to sign the contract
+	signingDuration time.Duration
+}
+
+// NewSigner returns an initialized employee identity contract signer
+func NewSigner(vcr vcr.VCR, publicURL string) contract.Signer {
+	return &signer{
+		// NewSessionStore returns an initialized SessionStore
+		store:           NewSessionStore(),
+		vcr:             vcr,
+		publicURL:       publicURL,
+		signingDuration: 10 * time.Minute,
+	}
+}
+
+// SigningSessionStatus returns the status of a signing session
+// If the session is completed, a VerifiablePresentation is created and added to the result
+// The session is deleted after the VerifiablePresentation is created, so the completed result can only be retrieved once
+func (v *signer) SigningSessionStatus(ctx context.Context, sessionID string) (contract.SigningSessionResult, error) {
+	s, ok := v.store.Load(sessionID)
 	if !ok {
 		return nil, services.ErrSessionNotFound
 	}
 	var vp *vc.VerifiablePresentation
+	issuerID, err := did.ParseDID(s.Employer)
+	if err != nil {
+		return nil, fmt.Errorf("invalid issuer DID: %w", err)
+	}
 
-	if s.status == SessionCompleted {
+	if s.Status == types.SessionCompleted {
```

Comment:
this requests a new VC each time this endpoint is called, really weird.

Reply:
Only when called in parallel, since it gets deleted. But I added an extra check.